### PR TITLE
fix: fix start sync publish event report

### PR DESCRIPTION
**Pull request by Han-Ya-Jun**
### Description
优化启动全量同步导致重复事件上报

Fixes # (issue)

### Checklist

- [x] 填写 PR 描述及相关 issue (write PR description and related issue)
- [x] 代码风格检查通过 (code style check passed)
- [x] PR 中包含单元测试 (include unit test)
- [ ] 单元测试通过 (unit test passed)
- [x] 本地开发联调环境验证通过 (local development environment verification passed)

### CODE REVIEW

File: pkg/agent/agent.go [hunk 1]
```diff
@@ -174,6 +175,11 @@ func (w *EventAgent) bootstrapSync(ctx context.Context) error {
 		return err
 	}
 
+	// 避免启动全量同步,导致大量publish event上报
+	for i := range stageList {
+		stageList[i].PublishID = constant.NoNeedReportPublishID
```
Comment (wklken) [inline pkg/agent/agent.go:180]: PublishID 为 0 才覆盖; 如果有值代表是正在发布的?

File: pkg/eventreporter/reporter.go [hunk 2]
```diff
@@ -159,22 +159,25 @@ func ReportLoadConfigurationDoingEvent(ctx context.Context, stage *v1beta1.BkGat
...
-	reporter.eventChain <- event
+	addEvent(event)
 }
 
 // ReportLoadConfigurationResultEvent Report the detection result of apisix loading
 func ReportLoadConfigurationResultEvent(ctx context.Context, stage *v1beta1.BkGatewayStage) {
+	// filter not need report event
+	publishID := stage.Labels[config.BKAPIGatewayLabelKeyGatewayPublishID]
+	if publishID == constant.NoNeedReportPublishID || publishID == "" {
+		logging.GetLogger().Infof("event[publish_id:%s] is not need to report", publishID)
```
Comment (zhu327) [inline pkg/eventreporter/reporter.go:170]: 打debug

Comment (wklken) [inline pkg/eventreporter/reporter.go:170]: 这个日志没有任何意义 => 打出来都一样, 应该加入gateway/stage相关信息

File: pkg/eventreporter/reporter.go [hunk 3]
```diff
@@ -230,6 +233,17 @@ func ReportLoadConfigurationResultEvent(ctx context.Context, stage *v1beta1.BkGa
 	})
 }
 
+// addEvent add event to reporter event
+func addEvent(event reportEvent) {
+	// filter not need report event
+	publishID := event.stage.Labels[config.BKAPIGatewayLabelKeyGatewayPublishID]
+	if publishID == constant.NoNeedReportPublishID || publishID == "" {
+		logging.GetLogger().Infof("event[publish_id:%s] is not need to report", publishID)
```
Comment (zhu327) [inline pkg/eventreporter/reporter.go:241]: 打debug

File: pkg/agent/agent.go [hunk 4]
```diff
@@ -328,9 +334,16 @@ func (w *EventAgent) handleTicker(ctx context.Context) {
 		if err != nil {
 			w.logger.Error(err, "List stage failed when all stage event triggered")
 			w.stageTimer.Update(registry.StageInfo{})
+
+			// 避免全量同步,导致大量publish event上报
+			for i := range stageList {
+				stageList[i].PublishID = constant.NoNeedReportPublishID
+			}
```
Comment (wklken) [inline pkg/agent/agent.go:341]: 同上

File: pkg/eventreporter/reporter.go (hunk 3 above)
Comment (Han-Ya-Jun) [reply pkg/eventreporter/reporter.go:241]: done

File: pkg/eventreporter/reporter.go (hunk 2 above)
Comment (Han-Ya-Jun) [reply pkg/eventreporter/reporter.go:170]: done

File: pkg/agent/agent.go (hunk 4 above)
Comment (Han-Ya-Jun) [reply pkg/agent/agent.go:341]: done

**Pull request merged**